pkg/ingester/wal/encoding.go

```diff
@@ -11,6 +11,11 @@ import (
 	"github.com/grafana/loki/pkg/util/encoding"
 )
 
+var (
+	// shared pool for WALRecords and []logproto.Entries
+	recordPool = NewRecordPool()
+)
+
```

Comment:
I think we don't need this package global here? it was just used in the actual ingester package, so we can create one with `NewRecordPool` there if we really need a global

Reply:
Yhea, just found the dependency why I created that in that file:
- It was used in encoding_test, so created one as global there for the tests themselves to use
- [Used in `Record#Reset`](https://github.com/grafana/loki/blob/main/pkg/ingester/encoding.go#L57). Inverted the dependency, and moved that to the pool itself, since it was the only caller of reset

Also, created one as global in `pkg/ingester/wal.go` since it's used a bunch there